src/EventStore.Core/TransactionLog/FileNamingStrategy/VersionedPatternFileNamingStrategy.cs

```diff
@@ -44,6 +45,73 @@ public class VersionedPatternFileNamingStrategy : IFileNamingStrategy {
 			return versions;
 		}
 
+		private int GetFileNumberFor(string path) {
+			var fileName = Path.GetFileName(path);
+			if (!_chunkNamePattern.IsMatch(fileName))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			var start = _prefix.Length;
+			var end = fileName.IndexOf('.', _prefix.Length);
+			Debug.Assert(end != -1);
+
+			if (!int.TryParse(fileName[start..end], out var chunkNumber))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			return chunkNumber;
+		}
+
+		private int GetFileVersionFor(string path) {
+			var fileName = Path.GetFileName(path);
+			if (!_chunkNamePattern.IsMatch(fileName))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			var dot = fileName.IndexOf('.', _prefix.Length);
+			Debug.Assert(dot != -1);
+
+			if (!int.TryParse(fileName[(dot+1)..], out var chunkVersion))
+				throw new ArgumentException($"Invalid file name: {fileName}");
+
+			return chunkVersion;
+		}
+
+		public void EnumerateAllFiles(
+			Func<string, int, int, int> getNextFileNumber,
```

Comment:
Consider either
1. Ensuring that these arguments are not null and passing empty actions, or
2. Assigning empty actions to them if they are null

So that they can be called more naturally `onFileMissing(..)` vs `onFileMissing?.Invoke(..)`

Reply:
Resolved in 858121198a22f1e879dfc95d9b7d628e0c1d288f